### PR TITLE
Ts move helpers packages

### PR DESCRIPTION
**Pull request by ethanfrey**
Migrate small packages to typescript:

- [x] util
- [x] sync
- [x] helper

I want to start on prom or action, but I noticed a circular dependency there, which should be resolved.

Importing `action` from `prom`: https://github.com/iov-one/bcp-redux/blob/master/src/prom/boot.mjs#L9-L19

Importing `prom` from `action`: https://github.com/iov-one/bcp-redux/blob/master/src/action/accountSample.mjs#L7-L11

Maybe boot needs to move from prom to action???

Also `selector` seems like a nice set of pure functions, only depending on the state, but even there is one import from `action`: https://github.com/iov-one/bcp-redux/blob/master/src/selector/transient/suggestAction.mjs#L10-L12

I like the architecture, good separation. Just seems a few pieces are importing from action when they shouldn't be (I mean, the code needs it, just think these need to be in a different package)

### CODE REVIEW

**Comment by Isabello** [comment]
To answer your question about boot, its already got a set of actions associated with it. We provide a full boot sequence that requires dispatching all 3 actions in order.

importing from from action is typical, as we need to tell the payload what promise to use.

**Comment by ethanfrey** [comment]
I understand why you import. But think...

action imports promise in many places
promise imports action here.

somehow this compiles/runs, but I am very wary of circular dependencies in code. in a static language it would fail to compile. here, it just seems like a poor location.

I would propose making a new package `higher` that contains these items that aren't "pure". Then we can have a nice hierarchy:

`higher(?)` (proposed) ->`reducer -> actions -> prom -> selector -> {sync, util, helper...}` (this dependency chain is basically the current design with 2-3 exceptions)

**Comment by Isabello** [comment]
Sure we can move the combined dispatch from `promCheckWithBoot` into its own package

File: ts/sync/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { bnsCodec } from "@iov/bns";
+import { Encoding } from "@iov/encoding";
+import { PublicKeyBundle } from "@iov/tendermint-types";
+
+// syncConvertIdentitiesToAddresses converts a list of identities into hex addresses
+// TODO: change args, as we ignore chainId
```
Comment (Isabello) [inline ts/sync/index.ts]: Chain Id will be required when we implement more chains that the BNS, as we need to use a different codec to get an address

**Comment by rudi-cilibrasi** [comment]
tree is best for understanding. so topo sort as per ethan is best.  we used to have rules "prom never depends on redux" so the prom that takes an action should move to a new dir and have a new suffix like `seq` or `sequence` to signify a sequence of redux actions. this term is also appropriate because like the semicolon "sequence point" each action makes visible a redux state transition to the rest of the system. so the term is apt for the redux state computer.

for suggestAction yes it can rename / move to something like strategy module.

glad to see you are recreating the genesis of original type theory in the Hilbert Programme for the last 120 years. the levels break cycles to prevent contradictions. It does indeed all come down to assigning levels / breaking cycles in type theory and this was true as much in the beginning as it is currently with Voevodsky's `h-level` in univalent foundations type theory, a major modern player.

**Pull request merged**